### PR TITLE
Naively ported to Swift 2.0

### PR DESCRIPTION
**Pull request by jpsim**
Biggest changes here are:

1. No longer needs to `Box`
2. Swapped protocol free-functions to the new stdlib's protocol extension versions
3. `Printable` -> `CustomStringConvertible`

### CODE REVIEW

File: Commandant/Switch.swift [hunk 1]
```diff
@@ -59,6 +59,6 @@ public func <| <ClientError> (mode: CommandMode, option: Switch) -> Result<Bool,
 		return .success(enabled)
 
 	case .Usage:
-		return .failure(informativeUsageError(option.description, option.usage))
```
Comment (robrix) [inline Commandant/Switch.swift]: Maybe we should remove the external label on the second parameter instead. I don’t have a strong opinion other than to note that `usage` is perhaps not a terribly helpful external label.

File: Commandant/Option.swift [hunk 2]
```diff
@@ -167,16 +167,16 @@ public func <*> <T, U, ClientError>(f: T -> U, value: Result<T, CommandantError<
...
-		return .failure(left.value)
+		return .failure(left)
 
 	case let (.Success, .Failure(right)):
-		return .failure(right.value)
+		return .failure(right)
 
 	case let (.Success(f), .Success(value)):
-		let newValue = f.value(value.value)
+		let newValue = f(value)
```
Comment (robrix) [inline Commandant/Option.swift]: yesssss

**Comment by jspahrsummers** [comment]
@jpsim Thanks for taking this on! :sparkles:

I'd like `master` to match the latest production version of Swift, so I don't think we want to pull this PR in as-is.

However, since it will probably be useful to bring in these changes (for SourceKitten, at the very least), I think maintaining a branch with Swift 2 support would be :gem:. To that end, I've added you as a collaborator—feel free to push your branch to the main repo. :bow:

**Comment by jpsim** [comment]
Makes total sense, I won't be merging Swift 2 support in my own repos until it goes GM either. I'll push up this branch soon. Thanks for the invite!

**Pull request closed without merging**